src/Api/Middleware/EventLogStorageCommitMiddleware.cs

```diff
@@ -5,18 +5,30 @@ namespace Passwordless.Api.Middleware;
 public class EventLogStorageCommitMiddleware
 {
     private readonly RequestDelegate _next;
+    private readonly ILogger<EventLogStorageCommitMiddleware> _logger;
 
-    public EventLogStorageCommitMiddleware(RequestDelegate next) => _next = next;
+    public EventLogStorageCommitMiddleware(RequestDelegate next, ILogger<EventLogStorageCommitMiddleware> logger)
+    {
+        _next = next;
+        _logger = logger;
+    }
 
-    public async Task InvokeAsync(HttpContext context, IEventLogger eventLogger)
+    public async Task InvokeAsync(HttpContext context, IServiceProvider provider)
```

Comment:
Why resolve the logger like this? Is it because the service registration can change?

Reply:
Since we're in middleware, the event logger that gets injected is outside the authentication context. this way, we can get the instance of the event logger that reflects the current state of the request.

Reply:
That took a while to understand, nice catch Jared.

Reply:
Added comment explaining why.